distributed/shuffle/_arrow.py

```diff
@@ -85,3 +78,40 @@ def deserialize_table(buffer: bytes) -> pa.Table:
 
     with pa.ipc.open_stream(pa.py_buffer(buffer)) as reader:
         return reader.read_all()
+
+
+def read_from_disk(path: Path, meta: pd.DataFrame) -> tuple[Any, int]:
+    import pyarrow as pa
+
+    batch_size = parse_bytes("1 MiB")
+    batch = []
+    shards = []
+    schema = pa.Schema.from_pandas(meta, preserve_index=True)
```

Comment:
Not using `pyarrow_schema_dispatch` here because it doesn't support `preserve_index` yet.